ui/ducks/send/helpers.js

```diff
@@ -133,27 +96,16 @@ export async function estimateGasLimitForSend({
   }
 
   try {
-    // Call into the background process that will simulate transaction
-    // execution on the node and return an estimate of gasLimit
-    const estimatedGasLimit = await estimateGas(paramsForGasEstimate);
-
-    const estimateWithBuffer = addGasBuffer(
-      estimatedGasLimit,
-      blockGasLimit,
+    const { gas } = await estimateGasBuffered(
+      paramsForGasEstimate,
       bufferMultiplier,
     );
-    return addHexPrefix(estimateWithBuffer);
+    return gas;
   } catch (error) {
-    const simulationFailed =
-      error.message.includes('Transaction execution error.') ||
-      error.message.includes(
-        'gas required exceeds allowance or always failing transaction',
-      ) ||
-      (CHAIN_ID_TO_GAS_LIMIT_BUFFER_MAP[chainId] &&
-        error.message.includes('gas required exceeds allowance'));
-    if (simulationFailed) {
+    if (error.simulationFails) {
```

Comment:
Can `estimateGasBuffered` even fail as it catches the query error and sets `simulationFails`?

If it did fail, where are the `error.gas` and `error.simulationFails` properties coming from?